From 1dd0eaad3861a8f7fb91e5dffc8ccc9c68576deb Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Fri, 30 Aug 2019 11:20:40 +0200
Subject: [PATCH] GCC: remove noexcept declaration of blink::OriginAccessEntry
 move constructor

blink::OriginAccessEntry move constructor is declared noexcept with default
implementation, but it depends on network::cors::OriginAccessEntry, that has
declared its move constructor without noexcept. That breaks compilation.

In this case we cannot make it noexcept too, as it contains const members
that cannot be converted to noexcept.

Bug: 819294
Change-Id: I9bc907279e2f1054f966b0dbcaaefb6d04596aa0

diff --git a/third_party/blink/renderer/platform/weborigin/origin_access_entry.cc b/third_party/blink/renderer/platform/weborigin/origin_access_entry.cc
index 2e9a9d1a7d2e..4c4d7ab77a26 100644
--- a/third_party/blink/renderer/platform/weborigin/origin_access_entry.cc
+++ b/third_party/blink/renderer/platform/weborigin/origin_access_entry.cc
@@ -59,7 +59,7 @@ OriginAccessEntry::OriginAccessEntry(
                network::mojom::CorsPortMatchMode::kAllowOnlySpecifiedPort,
                priority) {}
 
-OriginAccessEntry::OriginAccessEntry(OriginAccessEntry&& from) noexcept =
+OriginAccessEntry::OriginAccessEntry(OriginAccessEntry&& from) =
     default;
 
 network::cors::OriginAccessEntry::MatchResult OriginAccessEntry::MatchesOrigin(
diff --git a/third_party/blink/renderer/platform/weborigin/origin_access_entry.h b/third_party/blink/renderer/platform/weborigin/origin_access_entry.h
index b3e567474f6f..5113a406bd35 100644
--- a/third_party/blink/renderer/platform/weborigin/origin_access_entry.h
+++ b/third_party/blink/renderer/platform/weborigin/origin_access_entry.h
@@ -58,7 +58,7 @@ class PLATFORM_EXPORT OriginAccessEntry {
       network::mojom::CorsDomainMatchMode,
       network::mojom::CorsOriginAccessMatchPriority priority =
           network::mojom::CorsOriginAccessMatchPriority::kDefaultPriority);
-  OriginAccessEntry(OriginAccessEntry&& from) noexcept;
+  OriginAccessEntry(OriginAccessEntry&& from);
 
   network::cors::OriginAccessEntry::MatchResult MatchesOrigin(
       const SecurityOrigin&) const;
-- 
2.20.1

